app/src/test/java/org/mozilla/focus/telemetry/TelemetryWrapperTest.kt

```diff
@@ -0,0 +1,43 @@
+package org.mozilla.focus.telemetry
+
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.ArgumentMatchers.any
+import org.mockito.Mockito.spy
+import org.mockito.Mockito.times
+import org.mockito.Mockito.verify
+import org.mozilla.telemetry.Telemetry
+import org.mozilla.telemetry.TelemetryHolder
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.RuntimeEnvironment
+
+@RunWith(RobolectricTestRunner::class)
+class TelemetryWrapperTest {
+
+    private lateinit var telemetrySpy: Telemetry
+
+    @Before
+    fun setup() {
+        val telemetry = TelemetryFactory.createTelemetry(RuntimeEnvironment.application)
+        telemetrySpy = spy(telemetry)
+        TelemetryHolder.set(telemetrySpy)
+    }
+
+    @Test
+    fun `WHEN session events are called on TelemetryWrapper THEN associated Telemetry methods should be called`() {
+        TelemetryWrapper.startSession(RuntimeEnvironment.application)
+        TelemetryWrapper.stopSession(RuntimeEnvironment.application)
+        verify(telemetrySpy, times(1)).recordSessionStart()
+        verify(telemetrySpy, times(1)).recordSessionEnd(any())
+    }
+
+    /**
+     * See comment in [TelemetryWrapper.stopSession]
+     */
+    @Test
+    fun `WHEN TelemetryWrapper is called out of order THEN we should not crash`() {
```

Comment:
Should we additionally verify:
- when stopSession is called multiple times in a row (e.g. in theory, could happen in real world).
- That we create a telemetry event when this happens?

Reply:
Hm.  I can see how startSession could theoretically be called twice in a row (and I'll add that case), but I'm not sure how stopSession could be.

I'll check that telemetry events are being sent in this case too.